forc/src/ops/forc_init.rs

```diff
@@ -34,29 +34,38 @@ fn print_welcome_message() {
     );
 }
 
-pub fn canonicalize(path: &PathBuf) -> Result<String> {
-    match std::fs::canonicalize(path) {
-        Ok(p) => {
-            let as_str = String::from(p.to_string_lossy());
-            Ok(as_str)
-        }
-        Err(e) => anyhow::bail!("Could not derive canonical path: '{}'", e),
-    }
-}
-
 pub fn init(command: InitCommand) -> Result<()> {
     let project_dir = match &command.path {
         Some(p) => PathBuf::from(p),
-        None => std::env::current_dir().unwrap(),
+        None => {
+            std::env::current_dir().context("Failed to get current directory for forc init.")?
+        }
     };
 
+    if !project_dir.is_dir() {
+        anyhow::bail!("'{}' is not a valid directory.", project_dir.display());
+    }
+
+    if Path::exists(&Path::new(&project_dir).join("Forc.toml")) {
```

Comment:
```suggestion
    if project_dir.join(sway_utils::constants::MANIFEST_FILE_NAME).exists() {
```

Hot-tip: `PathBuf` automatically dereferences to `Path` (just like `String` derefs to `str`, or `Vec<T>` derefs to `[T]`), so there's no need to explicitly create a `Path` here.